[Lee Wei, David] iP

itsyme opened the pull request:
## Duke
> "A ceiling is simply a floor from below" - Elon Musk

Duke manages your tasks, it
* Keeps Track of Tasks (and their deadlines! :grin:)
*  Is text based
*  **SUPER** fast to use

All you have to do is,
1. Download it from [here](https://github.com/itsyme/ip/releases/tag/A-Jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you!

And it's *FREE*!

Features:
- [x] Managing Tasks
- [x] Managing Deadlines
- [ ] Reminders (coming soon!)

If you're a Java programmer, you can use it to practice Java too! Here is the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        new Duke("data/duke.txt").run();
    }
}
```

Commit:
Add Gradle support

itsyme committed:
Level-1

itsyme committed:
Level-2

itsyme committed:
Level-3

itsyme committed:
Level-4

itsyme committed:
A-TextUITesting

itsyme committed:
Add Level-5

itsyme committed:
Add Level-6

itsyme committed:
add Level-7

itsyme committed:
Add Level-8

itsyme committed:
Merge branch 'branch-Level-7'

itsyme committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

itsyme committed:
Fix Level-8

itsyme committed:
Add supporting files for A-MoreOOP

itsyme committed:
Add A-MoreOOP

itsyme committed:
Add A-Packages

itsyme committed:
Add A-JUnit

itsyme committed:
Add A-JavaDoc

itsyme committed:
Add A-CodingStandard

itsyme committed:
Add Level-9

itsyme committed:
Merge branch 'branch-A-JavaDoc'

itsyme committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Event.java
#	src/main/java/duke/Task.java

itsyme committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Parser.java

itsyme committed:
Fix Level-9

itsyme committed:
Update input for testing

itsyme committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

ryanongra commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,48 @@
+        this.ui = new Ui();
+        try {
+            this.tasks = new TaskList(this.storage.loadSave());
+        } catch (Exception e) {
```
Might be better to catch specific exceptions like IOException or FileNotFoundException?

ryanongra commented on src/test/java/duke/DeadlineTest.java:
```diff
@@ -0,0 +1,33 @@
+
+import org.junit.jupiter.api.Test;
+import java.time.LocalDate;
+import static org.junit.jupiter.api.Assertions.*;
```
Maybe should just import the assertEquals method?

ryanongra commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,66 @@
+
+    /**
+     * Returns date of <code>Event</code>
+     * @return Date
```
Very small thing but maybe can be a little more specific in the javadocs (e.g. returns the data that event is happening)

yewmeng28 replied:
Perhaps it would be better to be more descriptive like the `date` param in the `Event` constructor JavaDocs.

ryanongra commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,48 @@
+                command.execute(this.tasks, this.ui, this.storage);
+                this.storage.updateFile(this.tasks.getTasks());
+                isExit = command.isBye();
+            } catch (Exception e) {
```
Same as before, might be better to list the specific exceptions

ryanongra reviewed (commented):
LGTM! Just a few minor things :)

itsyme committed:
Add Level-10

itsyme committed:
Merge branch 'branch-Level-10'

yewmeng28 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,79 @@
+        this.ui = new Ui();
+        try {
+            this.tasks = new TaskList(this.storage.loadSave());
+        } catch (Exception e) {
```
Perhaps it would be better to specify the exceptions you would like to catch such as `IOException`.

yewmeng28 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,62 @@
+     * @return Done <code>Task</code>
+     */
+    Task setDone() {
+        return new Task(this.body, true);
```
Perhaps it would be better to simply set the `isDone` of the current `Task` to `true`.

yewmeng28 commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,42 @@
+     */
+    @Override
+    Task setDone() {
+        return new Todo(this.getBody(), true);
```
It would be easier to use the parent class's `setDone()` method (after changing it as per the previous comment) so you would not need to override this method in all the different task types.

yewmeng28 reviewed (commented):
LGTM! Only some minor tweaks to make your code look neater and make life easier for you :)

itsyme committed:
Add A-CheckStyle

itsyme committed:
Merge branch 'A-GradleCheckStyle'

itsyme committed:
Fix Gradle run

itsyme committed:
Add A-Assertions

Assertions absent.

Adding assertions allows us to document the correctness of assumptions in the code.

Let's add assertions to the important assumptions that needs to be held at various points in the code.

Using assertions is a way to document the correctness of assumptions in the code.

itsyme committed:
Add A-CodeQuality

Code quality is unchecked.

Code quality needs to be up to par to increase readability of the code.

Let's fix the code quality.

Using this code quality standard is provided by this module.

itsyme committed:
Merge pull request #1 from itsyme/branch-A-CodeQuality

Add A-CodeQuality

itsyme committed:
Add A-Streams

itsyme committed:
Merge pull request #2 from itsyme/branch-A-Assertions

Add A-Assertions

itsyme committed:
Merge branch 'master' into branch-A-Streams

itsyme committed:
Merge pull request #3 from itsyme/branch-A-Streams

Add A-Streams

itsyme committed:
Add BCD-Extension

Absence of help command.

Having a help command will help users navigate Duke more easily and intuitively.

Let's add a help command.

It is done by adding a help command as a help command is intuitive for users to ask for help.

itsyme committed:
Merge branch 'branch-BCD-Extension'

itsyme committed:
Add A-BetterGui

itsyme committed:
Merge branch 'Branch-A-BetterGui'

itsyme committed:
Add A-UserGuide
